nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/locale/LocaleProvider.java

```diff
@@ -60,4 +60,14 @@
      */
     public TimeZone getTimeZone(CoreSession session);
 
+    /**
+     * Returns the default locale for given requested locale.
+     * <p>
+     * Checks if requested locale is part of supported locales, and fall-backs on a default locale if not (or if
+     * requested locale is empty).
+     *
+     * @since 8.4
+     */
+    public Locale getDefaultLocale(String requestedLocale);
```

Comment:
Naming issue compared to the behavior and the existing methods:
- `getDefaultLocale()` vs existing `getLocale()` and `getLocaleWithDefault()`: the meaning of the new one is not obvious.
- `getDefaultLocale()` vs `Locale.getDefault()`: the naming suggests that both are equivalent.

The execution path also raises that naming issue:
- `getLocale(CoreSession) -> getLocaleWithDefault(CoreSession) -> getDefaultLocale() -> getDefaultLocale(String)`
- `getLocale(DocumentModel) -> getLocaleWithDefault(DocumentModel) -> getDefaultLocale() -> getDefaultLocale(String)`

Ideally, such a method would be named `getLocale(String)`.
Else: `getDefaultLocale()` does "get requested Locale or the default as fallback" => why not something like `getLocaleOrDefault()` ?